ansible/roles/docker/tasks/main.yml

```diff
@@ -25,3 +25,4 @@
 - name: install docker-py
   pip:
     name: docker-py
+  when: kubernetes_enable_cached_images | bool
```

Comment:
kubernetes_enable_cached_images comes from the kubernetes role and there is no dependency relationship between the two. In general I think it is best if we avoid having roles depend on variables that are defined and defaulted in other roles.

I think there are a couple of different ways to make this work without creating an implicit dependency between the two roles.
- Move the docker-py installation to the kubernetes role (removes the implicit dependency on the kubernetes role from the docker role, but still leaves an implicit dependency on the docker role from the kubernetes role)
- Add a new variable docker_install_docker_py to the docker role and add an explicit dependency on the docker role to the kubernetes role

The explicit dependency can be done using the meta/main.yml dependencies or by using include_role, both of which allow conditional dependencies.

Reply:
Good catch. I think I may just flat-out remove the functionality as it is not being used.

Reply:
Sounds good, there is work currently to add this functionality to kubeadm fwiw. https://github.com/kubernetes/kubeadm/issues/825